### PR TITLE
Input Handling in GVRf

### PR DESCRIPTION
**Pull request by rahul27**
This patch lays out the groundwork for input handling
in GVRf.

The introduced GVRInputManager allows user defined InputEvent
to be dispatched to through the scene graph. This enables
the GVRSceneObjects to implement user interaction using the
SensorEvents generated from GVRBaseSensor implementations.

For now the Framework only allows head tracking of a 2D pointer.

In the coming weeks I be posting samples on how to go about 
using this Framework.

GearVRf-DCO-1.0-Signed-off-by: Rahul Rudradevan
rahuldevan@gmail.com

### CODE REVIEW

File: GVRf/Framework/src/org/gearvrf/SensorManager.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            sensors.add(sensor);
+        }
+    }
+
+    void clear() {
+        sensors.clear();
+    }
+
+    void removeSensor(GVRBaseSensor sensor) {
+        if (sensors.contains(sensor) == false) {
```
Comment (rahul27) [inline GVRf/Framework/src/org/gearvrf/SensorManager.java]: Will fix this check, should be "==true"

File: GVRf/Framework/src/org/gearvrf/GVRInputManager.java [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+ */
+public class GVRInputManager {
+    private static final String TAG = GVRInputManager.class.getSimpleName();
+    private PointerSceneObject pointerSceneObject;
+    private final SensorManager sensorManager;
+
+    private ListenerDelegate listenerDelegate;
+
+    public interface InputEventListener {
+        public void onInputEvent(float[] values);
```
Comment (rahul27) [inline GVRf/Framework/src/org/gearvrf/GVRInputManager.java]: Will add javadocs for this interface ... and probably rename it to a more meaningful name.

**Comment by thomasflynn** [comment]
an initial scan of the code looks fine as far as i can tell.  as you point out, you still need javadocs in the appropriate places.  let me know when you have an update and i'll review again.

**Comment by rahul27** [comment]
Hi @thomasflynn, this commit is ready for review.

File: GVRf/IO/ControllerSample/assets/gvr_note4.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (thomasflynn) [inline GVRf/IO/ControllerSample/assets/gvr_note4.xml]: both this and the other one are using the old format.  in fact, you can get rid of the gvr_s5 one.  please copy the one from simple-sample.

File: GVRf/IO/ControllerSample/res/layout/activity_sample.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<!--/* Copyright 2015 Samsung Electronics Co., LTD
```
Comment (thomasflynn) [inline GVRf/IO/ControllerSample/res/layout/activity_sample.xml]: are this and fragment_sample used?

File: GVRf/IO/ControllerSample/res/menu/sample.xml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+<!--/* Copyright 2015 Samsung Electronics Co., LTD
```
Comment (thomasflynn) [inline GVRf/IO/ControllerSample/res/menu/sample.xml]: is this used?

File: GVRf/IO/STL/project.properties [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+#
+# To customize properties used by the Ant build system edit
+# "ant.properties", and override values to adapt the script to your
+# project structure.
+#
+# To enable ProGuard to shrink and obfuscate your code, uncomment this (available properties: sdk.dir, user.home):
+#proguard.config=${sdk.dir}/tools/proguard/proguard-android.txt:proguard-project.txt
+
+# Project target.
+target=android-22
```
Comment (thomasflynn) [inline GVRf/IO/STL/project.properties]: does this need to be 22?  or could it be 19?  if possible, i'd prefer not to lock-out kitkat users yet.

**Comment by thomasflynn** [comment]
also, you have a major package naming issue.  you have com.gearvrf.* rather than org.gearvrf.*  I'm not able to build the controller sample.

**Comment by rahul27** [comment]
This pull request is not ready for merge. I'll create a new one when done.

**Pull request closed without merging**